docs/evaluation/README.md

```diff
@@ -6,115 +6,303 @@ Evaluating Line-of-Business (LOB) chatbots presents unique challenges that tradi
-- **Function Call Arguments Recall Evaluator** – Checks if all required parameters are included ([function_call_recall.py](../../evaluation/chatbot/evaluators/function_call_recall.py))
-- **Function Call Reliability Evaluator** – Measures overall success in completing business processes ([function_call_reliability.py](../../evaluation/chatbot/evaluators/function_call_reliability.py))
+- **Function Call Precision Evaluator** (`FunctionCallPrecisionEvaluator`) – Measures if the chatbot calls the right functions without calling unnecessary ones ([function_call_precision.py](../../evaluation/chatbot/evaluators/function_call_precision.py))
+- **Function Call Recall Evaluator** (`FunctionCallRecallEvaluator`) – Assesses if all necessary functions are called ([function_call_recall.py](../../evaluation/chatbot/evaluators/function_call_recall.py))
+- **Function Call Arguments Precision Evaluator** (`FunctionCallArgsPrecisionEvaluator`) – Evaluates parameter accuracy and checks for unnecessary arguments in function calls ([function_call_precision.py](../../evaluation/chatbot/evaluators/function_call_precision.py))
+- **Function Call Arguments Recall Evaluator** (`FunctionCallArgsRecallEvaluator`) – Checks if all required parameters are included in function calls ([function_call_recall.py](../../evaluation/chatbot/evaluators/function_call_recall.py))
+- **Function Call Reliability Evaluator** (`FunctionCallReliabilityEvaluator`) – Measures overall success in completing business processes by combining function name and argument recall ([function_call_reliability.py](../../evaluation/chatbot/evaluators/function_call_reliability.py))
 
 Each evaluator produces a score between 0 and 1, with higher scores indicating better performance.
 
+## Key Metrics and Interpretation
+
+The framework provides five core metrics that together give a comprehensive view of chatbot performance:
+
+1. **Function Name Precision** (`Precision_fn`) - Measures if the chatbot calls only the functions it should call (no extra function calls)
+1. **Function Name Recall** (`Recall_fn`) - Measures if the chatbot calls all the functions it should call (no missing function calls)
+1. **Function Arguments Precision** (`Precision_args`) - Measures if function calls contain only the arguments they should contain (no extra arguments)
+1. **Function Arguments Recall** (`Recall_args`) - Measures if function calls contain all the arguments they should contain (no missing arguments)
+1. **Reliability Score** (`Reliability`) - Overall measure of task completion success, calculated as the mean of function name recall and argument recall
+
+**Interpretation Guidelines:**
+
+- **High Precision, Low Recall**: Chatbot is conservative but misses required actions
+- **Low Precision, High Recall**: Chatbot takes all required actions but also unnecessary ones
+- **High Reliability**: Chatbot successfully completes business tasks as intended
```

Comment:
As of now precision is not taken into account for calculating reliability. Would be worth explaining why this is enough to say that the Agent completes tasks successfully?